app/services/validator_check_active_service.rb

```diff
@@ -43,10 +43,13 @@ def should_be_destroyed?(validator)
   # number of previous by network
   def previous_epoch(network)
     @current_epoch ||= EpochWallClock.where(network: network).order(created_at: :desc).last
-    if network == 'testnet'
-      @previous_epoch_testnet ||= @current_epoch.epoch - 1
-    else
+    case network
+    when "mainnet"
       @previous_epoch_mainnet ||= @current_epoch.epoch - 1
+    when "testnet"
+      @previous_epoch_testnet ||= @current_epoch.epoch - 1
+    when "pythnet"
+      @previous_epoch_pythnet ||= @current_epoch.epoch - 1
```

Comment:
I suggest changing the method to be more generic something like that
```
def previous_epoch(network)
    @previous_epochs ||= {}
    return @previous_epochs[network] if @previous_epochs[network].present?

    current_epoch = EpochWallClock.where(network: network).order(created_at: :desc).last
    @previous_epochs[network] = current_epoch - 1
  end
```

Reply:
Thanks, I did apply that.